005-github-simplification.Rmd

```diff
@@ -0,0 +1,93 @@
+---
+output: github_document
+---
+
+# Tidyup 5: Github simplification
+
+Champion: Hadley Wickham\
+Co-champion:\
+Status: draft
+
+## Abstract
+
+## Motivation
+
+Originally, we divided tidyverse and r-lib because we thought it was important to keep data science and package development tools separate.
+Then when tidymodels kicked off, it seem natural for it to have its own home.
+
+But there are some downsides to having multiple organisations for the tidyverse team:
+
+-   While distinction between tidyverse, r-lib, and tidymodels is at reasonably strong in our heads, it's hard for users to articulate these differences, and to learn/remember exactly which organisation a package belongs to.
+    This is particularly hard when you look holistically across RStudio.
+
+-   There are a few packages that where it's less obvious which org they should belong too:
+
+    -   rvest: is it programming or data science? what about httr?
+    -   usethis: while it's technically programming, it's very user facing, and we expect data scientists might use it for analysis projects.
+    -   glue: useful for data science but designed to be zero dep specifically for use by package developers.
+    -   tibble/magrritr: both technically user-facing, but most people use via other packages
+    -   modelr: we don't really want people to use this package.
+    -   googledrive: not really about data analysis or package development.
+    -   vroom is in r-lib, readr is in tidyverse.
+
+    It often takes me multiple attempts to get to the correct URL for these packages.
+
+-   It's painful to move issues across orgs.
+    This affects packages like dplyr where the underlying issue might need to be fixed in vctrs.
+
+-   As we start to use GitHub teams more, managing separate teams across separate orgs will become more painful.
+    It also makes it hard to centralise discussions, and to explain succintly the scope of tidy ups.
+
+## Solutions
+
+There are three possible solutions:
+
+-   Move all r-lib and tidymodels repos to **tidyverse org**.
+-   Create a **new org** and move all r-lib, tidyverse, and tidymodel repos.
+-   Move all r-lib, tidyverse, and tidymodel to the **rstudio org**.
+
+These cases only affect the GitHub; all other websites would stay as is.
```

Comment:
Does that mean pkgdown websites like `vroom.r-lib.org` etc would continue? That seems weird once there is no longer an r-lib organization.

Reply:
Yeah, it'd be weird for the r-lib packages. I think we'd eventually move their websites elsewhere (maybe https://pkgs.rstudio.com?), but moving them is more fiddly since the redirects are harder, so I wanted to keep distinct from this tidyup.

Reply:
Will add some more text about this.